File: BS/soluciones/csharp/solve.cs
using System;
using System.IO;
using System.Numerics; // Necesario para BigInteger
using System.Diagnostics; // Necesario para Stopwatch

namespace SumaFibonacci
{
    class Program
    {
        static void Main(string[] args)
        {
            // Iniciamos un cronómetro
            Stopwatch stopwatch = new Stopwatch();
            stopwatch.Start();

            BigInteger f0 = 0;   // Fibonacci(0)
            BigInteger f1 = 1;   // Fibonacci(1)
            BigInteger suma = 0; // Acumulador para la suma

            // Iteramos 10.000 veces
            for (int i = 0; i < 10_000; i++)
            {
                // Sumamos el término actual de la serie
                suma += f0;

                // Calculamos el siguiente número de Fibonacci
                BigInteger siguiente = f0 + f1;
                f0 = f1;
                f1 = siguiente;
            }

            // Detenemos el cronómetro
            stopwatch.Stop();
            long tiempoTranscurridoMs = stopwatch.ElapsedMilliseconds;

            // Mostramos en consola
            Console.WriteLine($"La suma de los primeros 10,000 números de Fibonacci es: {suma}");
            Console.WriteLine($"Tiempo transcurrido: {tiempoTranscurridoMs} ms");

            // Escribimos los resultados en un archivo de texto
            string rutaArchivo = "output.txt";
            using (StreamWriter writer = new StreamWriter(rutaArchivo))
            {
                writer.WriteLine($"La suma de los primeros 10,000 números de Fibonacci es: {suma}");
                writer.WriteLine($"Tiempo transcurrido: {tiempoTranscurridoMs} ms");
            }

            Console.WriteLine($"\nResultados guardados en el archivo '{rutaArchivo}'.");
            Console.ReadLine(); // Para pausar la consola antes de cerrar
        }
    }
}

File: BS/soluciones/python/solve.py
import time

def suma_fibonacci(n):
    """
    Retorna la suma de los primeros n números de la serie de Fibonacci.
    """
    a, b = 0, 1
    total = 0

    for _ in range(n):
        total += a
        a, b = b, a + b
    
    return total

inicio = time.time()  # time.time() devuelve el tiempo actual en segundos

# Calculamos la suma de los primeros 10,000 números de Fibonacci
resultado = suma_fibonacci(10000)

# Medimos el tiempo al finalizar la ejecución de la función
fin = time.time()

# Calculamos el tiempo transcurrido en milisegundos
tiempo_ms = (fin - inicio) * 1000

# Mostramos por consola el resultado y el tiempo de ejecución
print("La suma de los primeros 10,000 términos de Fibonacci es:", resultado)
print(f"Tiempo de ejecución: {tiempo_ms:.2f} ms")

# Guardamos el resultado y el tiempo en un archivo .txt
with open("/app/BS/soluciones/python/", "w") as archivo:
    archivo.write(f"La suma de los primeros 10,000 términos de Fibonacci es: {resultado}\n")
    archivo.write(f"Tiempo de ejecución: {tiempo_ms:.2f} ms\n")
